stdlib/Random/src/deprecated.jl

```diff
@@ -37,3 +37,6 @@ end
 
 # PR #25668
 @deprecate RandomDevice(unlimited::Bool) RandomDevice(; unlimited=unlimited)
+
+# PR #28295
+@deprecate srand Random.seed false
```

Comment:
Add a deprecation for `guardsrand` -> `guardseed` as well? It was introduced in the 0.7 cycle but I know some packages use it so could be good to add.

Reply:
`guardsrand` was unexported in #27942, without deprecation...